frames/randomizer.py

```diff
@@ -47,6 +48,8 @@ def __init__(self, master=None):
 
         self.btnRandomize = ttk.Button(self, text='Randomize', command=self.doRandomize)
         self.btnRandomize.grid(column=2, row=5, padx=5, pady=5, sticky='nwse')
+        if not projects:
```

Comment:
Could maybe move the `self.entryName.current(0)` down here then wrap it in a try block? Then disable the button if it causes an exception and maybe display a popup that says no project or something?

Same for the pack frame

Reply:
I'm not sure causing and catching an exception is a good idea here because I can't find any documentation on it? It's a `_tkinter.TclError`, so it's probably internal. Showing a message box seems like a good idea. How about this:

```
        if projects:
            self.entryName.current(0)
        else:
            messagebox.showerror('No Data', 'No project data available. Please run the unpacker first.')
            raise RuntimeError("no project data available") # prevents the frame from changing
```